src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -167,18 +169,17 @@ public MgmtOutputLibrary()
                         if (count != 1)
                             continue;
 
-                        RequestPath requestPath = GetRequestPath(operationGroup, operation);
+                        RequestPath requestPath = RequestPath.FromOperation(operation, operationGroup);
                         var operationSet = RawRequestPathToOperationSets[requestPath];
                         var resourceDataModelName = ResourceDataSchemaNameToOperationSets.FirstOrDefault(kv => kv.Value.Contains(operationSet));
-                        var name = bodyParam.Schema.Language.Default.Name;
                         if (resourceDataModelName.Key is not null)
```

Comment:
This is very non-obvious though. `ResourceDataSchemaNameToOperationSets` doesn't have null keys, but `FirstOrDefault` may return default value for `KeyValuePair`, hence it is null. We need better code here.